pilot/pkg/model/service.go

```diff
@@ -324,6 +323,20 @@ type IstioEndpoint struct {
 	TLSMode string
 }
 
+// GetLocality returns the locality for the endpoint. If locality label is set we use this. Otherwise, we
+// use the one set by the registry:
+//   - k8s: region/zone, extracted from node's failure-domain.beta.kubernetes.io/{region,zone}
+// 	 - consul: defaults to 'instance.Datacenter'
+//
+// This is used by CDS/EDS to group the endpoints by locality.
+// TODO(nmittler): Why do we do this? Shouldn't the controller set the label appropriately in the endpoint?
```

Comment:
I am confused why we have a `Locality` field on the endpoint and then a `GetLocality` that doesn't just return that field? This seems either wrong or confusing, as noted by your comment

Reply:
This is preserving existing oddness. If we'd like to question why we do this now, I'm all for it. As my TODO suggests, I think the controllers should just set the locality field appropriately, so that we don't need this method. WDYT?

Reply:
Yeah I think we should get rid of it. If the controllers cannot set it, we should only have the method and not the field. But field is better if possible